cfgrammar/src/lib/yacc/parser.rs

```diff
@@ -334,13 +335,24 @@ impl YaccParser {
     }
 
     fn parse_action(&mut self, i: usize) -> YaccResult<(usize, String)> {
-        match RE_ACTION.find(&self.src[i..]) {
-            Some(m) => {
-                assert!(m.start() == 0 && m.end() > 0);
-                let s = self.src[i + 1..i + m.end() - 1].trim().to_string();
-                Ok((i + m.end(), s))
+        let mut j = i + 1; // Skip opening brace
+        let mut c = 1; // Count braces
+        while j < self.src.len() {
+            c += match self.src[j..].chars().next().unwrap() {
+                '{' => 1,
+                '}' => -1,
+                _ => 0
+            };
+            if c <= 0 {
+                break;
             }
-            None => Err(self.mk_error(YaccParserErrorKind::IllegalString, i))
+            j += 1;
```

Comment:
This isn't safe in unicode: characters can be more than 1 byte long. So you need to know which character you read above and increment the byte count appropriately here.